spec/controllers/admin/firms_controller_spec.rb

```diff
@@ -10,6 +10,12 @@ def create_user_with_firm(user_attrs, firm_attrs = {})
   end
 
   describe '#login_report' do
+    before do
+      allow_any_instance_of(FcaApi::Request)
+        .to receive(:get_firm)
+        .and_return(instance_double(FcaApi::Response, ok?: true))
+    end
+
```

Comment:
Noticing the amount of times and places we need to declare this `FcaApi` stub I wonder if we should stub it globally with a shared context. Something like this in the global RSpec config so it becomes a default:

```
RSpec.configure do |rspec|
  rspec.include_context "fca_api_ok_response", :include_shared => true
end
```
With this we would just need to override the default stub when we want to test the specific scenario when then API responds that the firm wasn't found.

Also think we can transform the stub to avoid the `allow_any_instance_of` that it tends to be discouraged.
I think something like this would achieve the same results (`let` declarations could be extracted if the stub inside stub approach became confusing):

```
before do
  allow(FcaApi::Request).to receive(:new).and_return(
    instance_double(
      FcaApi::Request, get_firm: instance_double(FcaApi::Response, ok?: true)
    )
  )
end
```

Didn't test any of this so I may be wrong, but hope the idea its clear. Let me know your thoughts :)